controllers/rhmi/rhmi_controller.go

```diff
@@ -434,6 +436,85 @@ func (r *RHMIReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 	return retryRequeue, err
 }
 
+func (r *RHMIReconciler) createSilence(installation *rhmiv1alpha1.RHMI) error {
+
+	host := "localhost:9093"
+	endpoint := "/api/v1/silences"
+
+	alertName := "KeycloakInstanceNotAvailable"
+
+	openshiftMonitoringNamespace := "openshift-monitoring"
+	openshiftPodName := "alertmanager-main-0"
+	containerName := "alertmanager"
+
+	integreatlyMonitoringNamespace := installation.Spec.NamespacePrefix + "middleware-monitoring-operator"
+	integreatlyPodName := "alertmanager-application-monitoring-0"
+
+	exists, err := silenceExists(openshiftMonitoringNamespace, openshiftPodName, host, endpoint, log, containerName, alertName)
+	if err != nil {
+		log.Error("err", err)
```

Comment:
Can we add a more descriptive error. Should we return at this point?

Reply:
Updated.